src/test/java/com/squareup/catalog/demo/example/RetrieveCatalogObjectExampleTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2017 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.catalog.demo.example;
+
+import com.squareup.catalog.demo.Logger;
+import com.squareup.connect.models.CatalogCategory;
+import com.squareup.connect.models.CatalogDiscount;
+import com.squareup.connect.models.CatalogItem;
+import com.squareup.connect.models.CatalogItemModifierListInfo;
+import com.squareup.connect.models.CatalogItemVariation;
+import com.squareup.connect.models.CatalogModifier;
+import com.squareup.connect.models.CatalogModifierList;
+import com.squareup.connect.models.CatalogObject;
+import com.squareup.connect.models.CatalogTax;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import static com.squareup.catalog.demo.util.Moneys.usd;
+import static com.squareup.connect.models.CatalogObject.TypeEnum.CATEGORY;
+import static com.squareup.connect.models.CatalogObject.TypeEnum.DISCOUNT;
+import static com.squareup.connect.models.CatalogObject.TypeEnum.ITEM;
+import static com.squareup.connect.models.CatalogObject.TypeEnum.ITEM_VARIATION;
+import static com.squareup.connect.models.CatalogObject.TypeEnum.MODIFIER;
+import static com.squareup.connect.models.CatalogObject.TypeEnum.MODIFIER_LIST;
+import static com.squareup.connect.models.CatalogObject.TypeEnum.TAX;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class RetrieveCatalogObjectExampleTest {
+
+  @Mock Logger logger;
+  private RetrieveCatalogObjectExample example;
+
+  @Before public void setUp() {
+    initMocks(this);
+    this.example = new RetrieveCatalogObjectExample(logger);
+  }
+
+  @Test public void logCatalogObjectDetails_category() {
+    CatalogObject catalogObject = catalogObject("id", CATEGORY)
+        .categoryData(new CatalogCategory()
+            .name("name"));
+    example.logCatalogObjectDetails(catalogObject, Collections.emptyMap());
+    verify(logger).info(anyString());
```

Comment:
I'm really just testing that it doesn't crash. If you make a mistake getting related objects and such, it's pretty easy to get an NPE.